index.html

```diff
@@ -3004,109 +3004,347 @@ <h3>Truncating or limiting the length of strings</h3>
-	</div>
-
+ 
+    
 	<div class="req" id="char_trunc_unit_rec">
 	<p class="advisement">Specifications that limit the length of a string SHOULD specify the length in terms of Unicode code points.</p>
 	</div>
+    
+    <p>Most limits are actually related to storage limitations, such as the size of a database field or a length limit in a protocol. Such limits are expressed in terms of [=code points=] in Unicode or [=code units=] (such as bytes) in a specific [=character encoding=]. Code points provides the best user experience, since all Unicode code points are treated identically: if text is truncated after 40 code points, all languages and scripts get the same number of code points to work with. By contrast, when the size limit is expressed in code units such as bytes in UTF-8, users who write in a language that mostly uses ASCII letters get many more characters (code points) for a given size limit than user's whose language is mostly made up of characters that take 2-, 3-, or 4-bytes per code point.</p>
+    
+    <div class="req" id="char_trunc_grapheme_boundary">
+	   <p class="advisement">Specifications that limit the length of a string SHOULD require truncation on [=visual text unit=] boundaries (usually as approximated by [=grapheme cluster=] boundaries), as truncation in the midst of a visual text unit such as a <a>combining character sequence</a> can alter the meaning of the string.</p>
+	</div>
+    
+    <aside style="border:1px solid green; padding: 5px" title="Important">
+        <p><q><em>Truncation on [=visual text unit=] boundaries</em></q> is not the same thing as using [=visual text units=] as the length limit.</p>
+    </aside> 
+    
+    <p>In addition to choosing how the length limit is expressed (code points vs. bytes), there is also the question of choosing the truncation boundary. Text must never be divided in the middle of a code point, because this results in corrupted characters. Text should never be divided in the middle of a [=grapheme cluster=] because this changes the appearance and meaning of the characters displayed.</p>
+
+    <aside class="example" title="Interaction of truncation and grapheme cluster boundaries" id="example-gc-cp-cu-trunc">
+		
+		<p>Returning to our Tamil example, here's the word for "Unicode", broken into syllables (which, remember, are treated as [=grapheme clusters=] in the Tamil script):</p>
+        
+        <table>
+            <tr>
+                <th>Syllable</th>
+                <td class="bigtext" style="width:6rem;text-align:center">&#x0BAF;&#x0BC2;</td>
+                <td class="bigtext" style="width:6rem;text-align:center">&#x0BA9;&#x0BBF;</td>
+                <td class="bigtext" style="width:6rem;text-align:center">&#x0B95;&#x0BCB;</td>
+                <td class="bigtext" style="width:6rem;text-align:center">&#x0B9F;&#x0BCD;</td>
+            </tr>
+            <tr>
+                <th>IPA</th>
+                <td style="width:6rem;text-align:center">juː</td>
```

Comment:
Move the dots again.

Reply:
If you want a word that contains both an inherent vowel and 'ko' in Tamil you could use this:

நன்கொடை
n̪an.ɡo.ɖai
donation

I found this by searching at https://r12a.github.io/scripts/taml/ta_vocab.html?q=%E2%92%B8%E2%92%B8.*%E0%AE%95%E0%AF%8A

See its composition at https://r12a.github.io/pickers/taml/?text=%E0%AE%A8%E0%AE%A9%E0%AF%8D%E0%AE%95%E0%AF%8A%E0%AE%9F%E0%AF%88 (click on List Characters)

Wiktionary entry is https://en.wiktionary.org/wiki/%E0%AE%A8%E0%AE%A9%E0%AF%8D%E0%AE%95%E0%AF%8A%E0%AE%9F%E0%AF%88#Tamil

Reply:
Aonther would be the following. This has no inherent vowel, but has a one code point initial vowel letter.

ஆலக்கொடி
aː.lak.ko.ɖi
yellow orpiment

https://en.wiktionary.org/wiki/%E0%AE%86%E0%AE%B2%E0%AE%95%E0%AF%8D%E0%AE%95%E0%AF%8A%E0%AE%9F%E0%AE%BF#Tamil

Reply:
TIL what 'yellow orpiment' is. 😸 

I don't really want a random word, if I can help it. I like using the word Unicode, in part because it is a transliterated word that should be well-known to spec writers (and fits in our world). It doesn't introduce every facet of Unicode. The only downside is that it introduces an urge to explain that Tamil doesn't always use pairs of code points. I've thought of a better way to explain this, which I'll add.

Reply:
This is the one change I didn't make, but I wrote a ton of new BPs and did more writing/reorg. Time for another look?